gflow/gflow.py

```diff
@@ -1,107 +1,255 @@
-import sys
 import json
 import logging
-from input import Input
-from config import Config
+import yaml
+import os
 from bioblend.galaxy.objects import *
-__version__ = "0.1.0"
 
 
 class GFlow(object):
-
-    def __init__(self, config):
-
+    def __init__(self, datadict):
+        """
+        Interact with a Galaxy instance
+
+        Args:
+            datadict (dict): The dictionary containing configuration parameters.
+        Attributes:
+            self.logger: For logging.
+            self.galaxy_url (str): The URL of an instance of Galaxy
+            self.galaxy_key (str): The API key of an instance of Galaxy
+            self.library_name (str): The name of the library to be created
+            self.history_name (str): The name of the history to be created
+            self.workflow_source (str): Whether the workflow's being imported from a file or with an id
+            self.workflow (str): Either a filename or id for the workflow
+            self.datasets_source (str): Whether the data's being imported from files or URLs
+            self.datasets (dict): A collection of filenames or URLs for the datasets
+            self.runtime_params (dict): A collection of required runtime parameters
+
+        """
         self.logger = logging.getLogger('gflow.GFlow')
-
-        self.options = config
-        self.datasets = []
-        for i in range(0, self.options.num_datasets):
-            self.datasets.append(Input(self.options.dataset_src, self.options.datasets[i], self.options.labels[i]))
+        try:
+            self.galaxy_url = datadict['galaxy_url']
+            self.galaxy_key = datadict['galaxy_key']
+            self.library_name = datadict['library_name']
+            self.history_name = datadict['history_name']
+            self.workflow_source = datadict['workflow_source']
+            self.workflow = datadict['workflow']
+        except KeyError as e:
+            self.logger.error("Missing required parameter: " + str(e))
+            raise KeyError
+        # Optional parameters follow
+        self.datasets_source = None
+        self.datasets = None
+        self.runtime_params = None
+        try:
+            self.datasets_source = datadict['datasets_source']
+            self.datasets = datadict['datasets']
+            self.runtime_params = datadict['runtime_params']
+        except KeyError as e:
+            self.logger.warning("%s parameter not set", e)
+
+    @classmethod
+    def from_config_file(cls, configfile):
```

Comment:
Names of methods should be name actions, not states. I.e. init_from_config_file. It makes for a longer name, but it is clear what you're doing when you're using the scrips.